deepchecks/nlp/text_data.py

```diff
@@ -275,6 +294,9 @@ def set_metadata(
         if self._metadata is not None:
             warnings.warn('Metadata already exist, overwriting it', UserWarning)
 
+        if isinstance(metadata, str):
+            metadata = pd.read_csv(metadata)
```

Comment:
Can we add a safeguard here, such that if the first column has the title 0 and has the entries 1..n then we raise warning that it's an index columns saved by mistake and we're ignoring it?

Reply:
Honestly, I don't think so. I think a user should be responsible for that themselves and they'll see their error.
Also, I don't really think anyone should use this, but added this for symmetry's sake as I added for properties.

Reply:
yeah I actually meant that for loading properties but can let this go.

Reply:
So for loading properties, if the user would originally use our load_properties function, then there won't be an issue to begin with :)